src/IECoreHoudini/GEO_CortexPrimitive.cpp

```diff
@@ -387,26 +424,75 @@ GA_PrimitiveTypeId GEO_CortexPrimitive::typeId()
 	return m_definition->getId();
 }
 
-#if UT_MAJOR_VERSION_INT >= 14
+#if UT_MAJOR_VERSION_INT >= 16
+
+void GEO_CortexPrimitive::create(
+	    GA_Primitive **newPrims,
+	    GA_Size numPrimitives,
+	    GA_Detail &detail,
+	    GA_Offset startOffset,
+	    const GA_PrimitiveDefinition &def )
+{
+
+	// allocate all the points and vertices at the same time
+	GA_Offset pointBlock = detail.appendPointBlock( numPrimitives );
+
+	if ( numPrimitives >= 4*GA_PAGE_SIZE )
+	{
+		// Allocate them in parallel if we're allocating many.
+		// This is using the C++11 lambda syntax to make a functor.
+		UTparallelForLightItems(UT_BlockedRange<GA_Offset>( startOffset, startOffset + numPrimitives ),
+				[ newPrims, &detail, startOffset, &pointBlock ]( const UT_BlockedRange<GA_Offset> &r ){
+			GA_Offset primOffset( r.begin() );
+			GA_Primitive **pprims = newPrims + ( primOffset - startOffset );
+			GA_Offset endOffset( r.end() );
+			for ( ; primOffset != endOffset; ++primOffset, ++pprims, ++pointBlock )
+			{
+				GEO_CortexPrimitive * newPrim = new GEO_CortexPrimitive( &detail , primOffset );
+				GA_Offset vertex = newPrim->allocateVertex( pointBlock );
+				newPrim->myVertexList.setTrivial( vertex, 1 );
+				*pprims = newPrim;
+			}
+		});
+	}
+	else
+	{
+		// Allocate them serially if we're only allocating a few.
+		GA_Offset endOffset( startOffset + numPrimitives );
+		for ( GA_Offset primOffset( startOffset ); primOffset != endOffset; ++primOffset, ++newPrims, ++pointBlock )
+		{
+			GEO_CortexPrimitive * newPrim = new GEO_CortexPrimitive( &detail , primOffset );
+			GA_Offset vertex = newPrim->allocateVertex( pointBlock );
+			newPrim->myVertexList.setTrivial( vertex, 1 );
+			*newPrims = newPrim;
+		}
+	}
+}
+
+#elif UT_MAJOR_VERSION_INT >= 14
 
 GA_Primitive *GEO_CortexPrimitive::create( GA_Detail &detail, GA_Offset offset, const GA_PrimitiveDefinition &definition )
 {
-	return new GEO_CortexPrimitive( static_cast<GU_Detail *>( &detail ), offset );
+	return new GEO_CortexPrimitive( &detail, offset );
```

Comment:
Actually...what will be the effect of changing this to GA_Detail? I think this means that the GEO_CortexPrimitive constructor with the GEO_Detail parameter will be called instead of the GA_Detail constructor. Should I remove the GA_Detail constructor then?